UDV-Core/src/Utils/3DTiles/3DTilesUtils.js

```diff
@@ -0,0 +1,186 @@
+import * as THREE from '../../../node_modules/three/src/Three.js';
+
+/**
+ * Search a batch table in a tile. A tile is a THREE.js 3DObject with a 
+ * 3-level hierarchy : Object3D > Scene > Mesh. This function searches into the
+ * for the batch table (which is located in the Object3D level).
+ * 
+ * @param {*} tile A 3DTiles tile object from THREE.js.
+ */
+export function getBatchTableFromTile(tile) {
+  if (!!tile.batchTable) {
+    return tile.batchTable;
+  } else if (!!tile.parent) {
+    return getBatchTableFromTile(tile.parent);
+  }
+  return undefined;
+}
+
+/**
+ * Gets an object batch ID from an intersection. This methods takes one of the
+ * 3 points of the intersection triangle and retrieves the corresponding batch
+ * ID in the intersection tile.
+ * 
+ * @param {*} inter An intersection
+ */
+export function getBatchIdFromIntersection(inter) {
+  let index = inter.face.a;
+  return inter.object.geometry.attributes._BATCHID.array[index];
+}
+
+/**
+ * Get the first intersection object where the target is a 3D object with
+ * a batch id (a.k.a. the "Mesh" of a tile).
+ * 
+ * @param {Array<any>} intersects The array of intersections, provided by
+ * itowns.View.pickObjectsAt
+ */
+export function getFirstTileIntersection(intersects) {
+  for (let inter of intersects) {
+    let geomAttributes = inter.object.geometry.attributes;
+    if (!!geomAttributes && !!geomAttributes._BATCHID) {
+      return inter;
+    }
+  }
+  return undefined;
+}
+
+/**
+ * Counts the number of 3DTiles tiles displayed by the view.
+ * 
+ * @param {*} layer The 3DTiles layer.
+ */
+export function getVisibleTileCount(layer) {
+  let tsroot = layer.object3d.children[0];
+  return Object.keys(tsroot.children).length;
```

Comment:
Use the length of tileindex.index instead (see the other comment about this).

Reply:
We don't want to to that here because tileIndex contains all tiles of the tileset, even the one that are not loaded in the scene. The purpose of this method is to count how many tiles are currently rendered.

Reply:
Ah ok my bad. However, this will still not work for another indexing method such as binary trees, quadtrees, etc.  Hence you might have to recurse on the object3D structure to find out the number of rendered tiles.

Reply:
Ok, I'm changing the method

Reply:
Done. I added a method `getVisibleTiles` that explores the tileset tree (by doing a DFS) and returns all tiles in an array. `getVisibleTileCount` now just returns the length of this array.